### PR TITLE
Mistral 7B Finetuning with QLora

### PR DESCRIPTION
**Pull request by thomaslemoullec**
Need to fix the PEFT Train docker image for the finetuning step.  Merging works correctly with the docker image, You might face an error when merging if the peft version used for merging is different from the one used for finetuning (new field added in the config_adapter file)

**REQUIRED:** Add a summary of your PR here, typically including why the change is needed and what was changed. Include any design alternatives for discussion purposes.

<br>
Implementation of QLora Finetuning in a separate notebook. Mixtral finetuning notebook will follow soon so we can split Mistral 7B and Mixtral. 
<br><br><br>

**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [ ] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [ ] Follow the style and grammar rules outlined in the above notebook template.
- [ ] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [ ] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [ ] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [ ] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

<br>

2. If you are opening a PR for `Community Notebooks` under the [notebooks/community](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/community) folder:
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/community/CODEOWNERS) file under the `Community Notebooks` section, pointing to the author or the author's team.
- [ ] Passes all the required formatting and linting checks. You can locally test with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).

<br>

3. If you are opening a PR for `Community Content` under the [community-content](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/community-content) folder:
- [ ] Make sure your main `Content Directory Name` is descriptive, informative, and includes some of the key products and attributes of your content, so that it is differentiable from other content
- [ ] The main content directory has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/community-content/CODEOWNERS) file under the `Community Content` section, pointing to the author or the author's team.
- [ ] Passes all the required formatting and linting checks. You can locally test with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).

### CODE REVIEW

**Review by thomaslemoullec — COMMENTED** [review]
Updated the docker peft to 20240126_0936_RC00

File: notebooks/community/model_garden/model_garden_pytorch_mistral_peft_tuning.ipynb [hunk 1]
```diff
@@ -0,0 +1,1460 @@
+{
```
Comment (gericdong) [inline notebooks/community/model_garden/model_garden_pytorch_mistral_peft_tuning.ipynb:1]: <p>Update the license year to 2024.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2645/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (gericdong) [inline notebooks/community/model_garden/model_garden_pytorch_mistral_peft_tuning.ipynb:1]: <p>Follow the notebook template to add missing sections including:</p><ul><li>Links to Colab, Workbench and Github</li><li>Overview, Objective, Dataset/Model, Costs</li><li>Installation (e.g. Vertex AI SDK, etc.)</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2645/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (gericdong) [inline notebooks/community/model_garden/model_garden_pytorch_mistral_peft_tuning.ipynb:1]: <p>Initiatialise &gt; Initialise</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2645/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (gericdong) [inline notebooks/community/model_garden/model_garden_pytorch_mistral_peft_tuning.ipynb:1]: <p>Use&nbsp;<a href="https://developers.google.com/style/capitalization#capitalization-in-titles-and-headings" target="_blank">sentence case to capitalize titles and headings</a>, Define Common Functions &gt; Define common functions</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2645/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (gericdong) [inline notebooks/community/model_garden/model_garden_pytorch_mistral_peft_tuning.ipynb:1]: <p>Move the section to the top of the notebook where the setup section is.</p><p> </p><p>Address the reader as you: <a href="https://developers.google.com/style/person" target="_blank">https://developers.google.com/style/person</a>: we &gt; you</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2645/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (gericdong) [inline notebooks/community/model_garden/model_garden_pytorch_mistral_peft_tuning.ipynb:1]: <p>Clear all output</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2645/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (gericdong) [inline notebooks/community/model_garden/model_garden_pytorch_mistral_peft_tuning.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.    # Tranformers provides all API for downloading and working with pre-trained models that are in the HF hub.
</pre><p>Tranformers &gt; Transformers</p><p>huggeringface &gt; Huggeringface</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2645/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (gericdong) [inline notebooks/community/model_garden/model_garden_pytorch_mistral_peft_tuning.ipynb:1]: <p>Use&nbsp;<a href="https://developers.google.com/style/tense" target="_blank">present tense</a> and address the reader as you: We'll &gt; You</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2645/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (gericdong) [inline notebooks/community/model_garden/model_garden_pytorch_mistral_peft_tuning.ipynb:1]: <p>Use&nbsp;<a href="https://developers.google.com/style/tense" target="_blank">present tense</a> and address the reader as you: We &gt; You, our &gt; your</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2645/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (gericdong) [inline notebooks/community/model_garden/model_garden_pytorch_mistral_peft_tuning.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.    # model = AutoModelForCausalLM.from_pretrained(model_id, quantization_config=bnb_config, device_map="auto")
</pre><p>Do we need to keep the dead code? If so, it's better to explain </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2645/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (gericdong) [inline notebooks/community/model_garden/model_garden_pytorch_mistral_peft_tuning.ipynb:1]: <p>Branding: Vertex &gt; Vertex AI</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2645/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

Comment (gericdong) [inline notebooks/community/model_garden/model_garden_pytorch_mistral_peft_tuning.ipynb:1]: <p>We'll &gt; You</p><p>our - your</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2645/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

Comment (gericdong) [inline notebooks/community/model_garden/model_garden_pytorch_mistral_peft_tuning.ipynb:1]: <p>Use&nbsp;<a href="https://developers.google.com/style/tense" target="_blank">present tense</a> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2645/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='42'/>

Comment (gericdong) [inline notebooks/community/model_garden/model_garden_pytorch_mistral_peft_tuning.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.    model = prepare_model_for_kbit_training(model)  # TODO: Explain
</pre><p>TODO?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2645/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>

Comment (gericdong) [inline notebooks/community/model_garden/model_garden_pytorch_mistral_peft_tuning.ipynb:1]: <p>Use&nbsp;<a href="https://developers.google.com/style/tense" target="_blank">present tense</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2645/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='52'/>

Comment (gericdong) [inline notebooks/community/model_garden/model_garden_pytorch_mistral_peft_tuning.ipynb:1]: <p>Adapater &gt; Adapter</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/2645/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='58'/>

**Pull request merged**